### ISSUE TITLE
[@auto-it/npm] Invalid option `--registry` passed to `lerna version`

### ISSUE THREAD

**Issue by AndrewLeedham** [intuit/auto#2129]
**Describe the bug**

When upgrading to `lerna@4.0.0` they seem to be more strict about valid options. `lerna version` as called in auto therefore fails as `--registry` is an option of `lerna publish` but not `lerna version`.

**To Reproduce**

- Upgrade from `lerna@3.x` to `lerna@4.0.0` in a monorepo project.
- Run `auto shipit`.

**Expected behavior**

In theory `lerna@4.0.0` should be a clean upgrade, but because Auto is using an invalid option it fails.

**Environment information:**

```txt
"auto" version: v10.32.5
"git"  version: v2.30.1 (Apple Git-130)
"node" version: v14.18.1
```

**Additional context**

Error log:
```
Error: Running command 'npx' with args [lerna, version, major, --registry, https://registry.npmjs.org/, --yes, --no-push, --no-git-tag-version, --no-commit-hooks, --exact, --ignore-scripts] failed

ERR! lerna Unknown argument: registry
```

Code that adds the `--registry` flag seems to be here: https://github.com/intuit/auto/blob/main/plugins/npm/src/index.ts#L941

**Comment by AndrewLeedham** [comment]
I found a temporary workaround by specifying the registry as `https://registry.npmjs.org` in my lerna config. Since there is a check against the default which stops it from being added.

**Comment by FlomoN** [comment]
This workaround doesnt seem to work anymore, but i don't understand why.
I also tried adding (with and without trailing slash) to my `lerna.json`.

```
"command": {
    "publish": {
      "registry": "https://registry.npmjs.org"
    }
  }
}
```

Also in package.json

```
"publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
```

Still when I run `npx auto shipit -d` I get

```
‼  warning   lerna notice cli v5.5.0
lerna info versioning independent
lerna info Assuming all packages changed
lerna success found 2 packages ready to publish

‼  warning   lerna notice cli v5.5.0
lerna info versioning independent
lerna success found 2 packages

Error: Running command 'npx' with args [lerna, version, prepatch, --registry, https://registry.npmjs.org/, --yes, --no-push, --no-git-tag-version, --no-commit-hooks, --exact, --ignore-scripts, --preid, canary.523b477] 
failed

ERR! lerna Unbekanntes Argument: registry
```

Any idea?

**Comment by AndrewLeedham** [comment]
The defaulting code still seems to be there: https://github.com/intuit/auto/blob/0c96b567e6e3c3e62797e12f92fc5d8e0c76617b/plugins/npm/src/index.ts#L250-L255

The registry is parsed as so, so either should work:
https://github.com/intuit/auto/blob/0c96b567e6e3c3e62797e12f92fc5d8e0c76617b/plugins/npm/src/set-npm-token.ts#L16-L30

The default registry is `"https://registry.npmjs.org"` https://github.com/intuit/auto/blob/0c96b567e6e3c3e62797e12f92fc5d8e0c76617b/plugins/npm/src/set-npm-token.ts#L13

It seems to still be working for me, so I can only assume is isn't detecting the config for some reason?

**Comment by FlomoN** [comment]
Maybe its something about the `--dry-run` flag? When i run it without, it seems to work as expected.

**Comment by Webbanditten** [comment]
I'm having the same issue. Running --dry-run results in `Unknown argument: registry`

**Comment by uschtwill** [comment]
Observing the same behaviour.

**Comment by ryantang333** [comment]
For anyone still experiencing this issue, downgrading lerna to 3.0.0 is a workaround.

**Comment by hipstersmoothie** [comment]
Oooo this seems pretty bad. I hand't noticed this in their changelogs. Would someone experiencing the issue want to make a PR? I'll be fast about merging

**Comment by hipstersmoothie** [comment]
Awesome!

**Comment by jackw** [comment]
I've just hit this issue whilst trying to debug why our repo started creating odd changelogs and unintentionally published a major version. We're running `auto@11.0.4` and `lerna@8.0.0`.

Unfortunately it was later removed again in [this PR](https://github.com/lerna/lerna/pull/3809) which according to the [commit in main](https://github.com/lerna/lerna/commit/73b81c354770d6e8c7ee43afe3840076bb27399a) was released in v7.2.0.

@hipstersmoothie I've opened a [PR here](https://github.com/intuit/auto/pull/2421) that removes the flag from all version commands that are wrapped in dry-run checks.
